src/hooks/useSelectedRevisions.ts

```diff
@@ -0,0 +1,34 @@
+import { setSelectedRevisions } from '../reducers/SelectedRevisionsSlice';
+import { RevisionsList } from '../types/state';
+import { useAppDispatch, useAppSelector } from './app';
+
+const useSelectRevision = () => {
+  const dispatch = useAppDispatch();
+
+  const baseCheckedRevisions: RevisionsList[] = useAppSelector(
+    (state) => state.search.base.checkedRevisions,
+  );
+
+  const newCheckedRevisions: RevisionsList[] = useAppSelector(
+    (state) => state.search.new.checkedRevisions,
+  );
+
+  const selectedRevisions = useAppSelector(
+    (state) => state.selectedRevisions.revisions,
+  );
+
+  const addSelectedRevisions = () => {
+    const newSelected = [...selectedRevisions];
+    newSelected.push(...baseCheckedRevisions, ...newCheckedRevisions);
+    const filteredSelected = newSelected.filter(
+      (revision, index, self) =>
+        self.findIndex((r) => r.id === revision.id) === index,
+    );
+
+    dispatch(setSelectedRevisions({ selectedRevisions: filteredSelected }));
```

Comment:
I wonder if the action could rather be `addSelectedRevisions` and would handle the push. Then you don't need the `selectedRevisions` selector above. I feel like this is a more correct way of working with the reducer.

Also I see you removed `clearCheckedRevisions` like I suggested but I don't see the reducer change to clear these values. Is that not needed anymore?

Reply:
Yes, since the checked revisions will only render on the search page, I don't need to clear them. 

I'll revisit executing this action as `addSelectedRevisions` in the `SearchSlice` reducer in the next patch.